pkg/policy/api/cidr.go

```diff
@@ -42,3 +48,41 @@ type CIDRRule struct {
 	// or provided by user
 	Generated bool `json:"-"`
 }
+
+// CIDRSlice is a slice of CIDRs. It allows receiver methods to be defined for
+// transforming the slice into other convenient forms such as
+// EndpointSelectorSlice.
+type CIDRSlice []CIDR
+
+// CIDRRuleSlice is a slice of CIDRRules. It allows receiver methods to be
+// defined for transforming the slice into other convenient forms such as
+// EndpointSelectorSlice.
+type CIDRRuleSlice []CIDRRule
+
+// ComputeResultantCIDRSet converts a slice of CIDRRules into a slice of
+// individual CIDRs. This expands the cidr defined by each CIDRRule, applies
+// the CIDR exceptions defined in "ExceptCIDRs", and forms a minimal set of
+// CIDRs that cover all of the CIDRRules.
+func ComputeResultantCIDRSet(cidrs CIDRRuleSlice) CIDRSlice {
+	var allResultantAllowedCIDRs CIDRSlice
+	for _, s := range cidrs {
+		// No need for error checking, as CIDRRule.Sanitize() already does.
+		_, allowNet, _ := net.ParseCIDR(string(s.Cidr))
```

Comment:
It might be worth noting that we expect perfect input in the function doc. We will, inevitably, call this with unsanitized addresses some day.